src/components/todo/TodoItem.tsx

```diff
@@ -49,9 +51,20 @@ const TodoItem = ({ todoItem, setLoading }: Props) => {
   }, [todoItem]);
 
   const [todo, setTodo] = useState(todoData);
-  const deleteTaskHandeler = () => {
-    dispatch(deleteTodoAction(todoItem));
+  const deleteTaskHandeler = async () => {
     setShowModal(false);
+    setLoading(true);
+    await axiosInstance.delete(`/todo/${todoItem.id}`).then(async () => {
+      await axiosInstance
+        .get("/todos")
```

Comment:
This looks bit mixed up, we can separate it for better readability.

Reply:
Updated